src/lib/generator/sitemapXml.js

```
@@ -0,0 +1,261 @@
+ *
+ * @param {string} baseUrl - The base URL for link normalization.
+ * @param {string} buffer - The buffer string to process for links.
+ * @returns {string[]} An array of extracted links.
+ */
+function processBuffer(baseUrl, buffer) {
+  const linkRegex = /<a\s+(?:[^>]*?\s+)?href=(["']?)([^"'\s>]+)\1/g;
+
+  const links = [];
+
+  let match;
+  while ((match = linkRegex.exec(buffer)) !== null) {
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)